zips/zip-0231.md

```diff
@@ -189,23 +217,31 @@ memos:
-- Each time decryption succeeds for a memo chunk, increment `counter` by 1, and
-  then continue attempting to decrypt subsequent chunks.
-- Once all memo chunks have been trial-decrypted once, set `final_chunk = 0x01`
-  and then attempt to decrypt the memo chunks again, starting immediately after
-  the last successfully-decrypted chunk (or at the start if none were), and
-  without incrementing `counter`.
-  - This step can be made secret-independent by attempting to decrypt every memo
-    chunk again, and ignoring the results of all chunks up to and including the
-    last successfully-decrypted chunk.
-- If no memo chunk decrypts successfully with `final_chunk = 0x01`, discard any
-  memo chunks that were decrypted, and return nothing. Otherwise, concatenate
-  the decrypted memo chunks in order and return the concatenation as the memo.
-
-If any chunk of the memo encrypted to `memo_key` has been pruned, the decryption
-process above returns nothing (as `final_chunk` will be set to `0x01` with the
+$\hspace{1.5em}$ let mutable $\mathsf{memo} \;{\small ⦂}\; \mathbb{N} \leftarrow 0 \\$
+$\hspace{1.5em}$ let mutable $\mathsf{counter} \;{\small ⦂}\; \mathbb{N} \leftarrow 0 \\$
+$\hspace{1.5em}$ let mutable $\mathsf{potential\_last\_chunk} \;{\small ⦂}\; \mathbb{N} \leftarrow 0 \\$
+$\hspace{1.5em}$ for $i$ from $0$ up to $\mathsf{len}(\mathsf{memo\_chunks})-1$: $\\$
+$\hspace{3.0em}$     if $\mathsf{memo\_chunks}[i]$ is not pruned: $\\$
+$\hspace{4.5em}$         let $\mathsf{nonce} = \mathsf{I2BEOSP}_{88}(\mathsf{counter}) \,||\, [\mathtt{0x00}] \\$
+$\hspace{4.5em}$         let $P = \mathsf{IETF\_AEAD\_CHACHA20\_POLY1305.Decrypt}(\mathsf{encryptionKey}, \mathsf{nonce}, \mathsf{memo\_chunks}[i]) \\$
+$\hspace{4.5em}$         if $P \neq \bot$: $\\$
+$\hspace{6.0em}$             set $\mathsf{memo} \leftarrow \mathsf{memo} \,||\, P \\$
+$\hspace{6.0em}$             set $\mathsf{counter} \leftarrow \mathsf{counter} + 1 \\$
+$\hspace{6.0em}$             set $\mathsf{potential\_last\_chunk} \leftarrow i + 1 \\$
+$\,\\$
+$\hspace{1.5em}$ let $\mathsf{nonce} = \mathsf{I2BEOSP}_{88}(\mathsf{counter}) \,||\, [\mathtt{0x01}] \\$
+$\hspace{1.5em}$ let mutable $\mathsf{success} \leftarrow \kern0.05em$ false $\\$
+$\hspace{1.5em}$ for $i$ from $0$ up to $\mathsf{len}(\mathsf{memo\_chunks})-1$: $\\$
+$\hspace{3.0em}$     if $\mathsf{memo\_chunks}[i]$ is not pruned: $\\$
+$\hspace{4.5em}$         let $P = \mathsf{IETF\_AEAD\_CHACHA20\_POLY1305.Decrypt}(\mathsf{encryptionKey}, \mathsf{nonce}, \mathsf{memo\_chunks}[i]) \\$
+$\hspace{4.5em}$         if $i \geq \mathsf{potential\_last\_chunk}$ and $P \neq \bot$ and not $\mathsf{success}$: $\\$
+$\hspace{6.0em}$             set $\mathsf{memo} \leftarrow \mathsf{memo} \,||\, P \\$
+$\hspace{6.0em}$             set $\mathsf{success} \leftarrow \kern0.05em$ true $\\$
+$\,\\$
+$\hspace{1.5em}$ if $\mathsf{success}$ then return $\mathsf{memo}$ else return $\bot \\$
```

Comment:
This renders strangely in github: 
![image](https://github.com/user-attachments/assets/1ce9585a-3732-46cd-86fb-57acef9718e4)